solutions/rust/04-fs_isolation/code/src/main.rs

```diff
@@ -0,0 +1,74 @@
+use anyhow::{Context, Result};
+use std::env::{args, set_current_dir};
+use std::fs::{copy, create_dir, create_dir_all, set_permissions, File, Permissions};
+use std::os::unix::fs::{chroot, PermissionsExt};
+use std::process::{exit, Command, Stdio};
+use tempfile::TempDir;
+
+// Usage: your_docker.sh run <image> <command> <arg1> <arg2> ...
+fn main() -> Result<()> {
+    let args: Vec<_> = args().collect();
+    let command = &args[3];
+    let command_args = &args[4..];
+
+    let exit_code = run_child(command, command_args)?;
+    exit(exit_code);
+}
+
+fn run_child(command: &String, command_args: &[String]) -> Result<i32> {
+    // Need the destructor to run so the directory is removed after use. See https://docs.rs/tempfile/3.3.0/tempfile/struct.TempDir.html#resource-leaking
+    let temp_dir = tempfile::tempdir()?;
+
+    // Don't want '/usr/local/bin/docker-explorer' sending us back to the root of the file system.
```

Comment:
@ThetaSinner I find this comment a bit confusing - what does "send back to the root of the file system" mean?